VisualRadio/app/src/main/java/altus/visualradio/MainListingActivity.java

```diff
@@ -0,0 +1,142 @@
+package altus.visualradio;
+
+import android.app.Activity;
+import android.app.ListActivity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ListView;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import altus.visualradio.models.ListDetailSetter;
+
+
+public class MainListingActivity extends ListActivity {
+    private static      String index_filename = "VS_index_feed.json";
+    private             ArrayList<ListDetailSetter> index_detail_setter;
+    private             JSONObject json_object;
+
+    private static      String JSON_INDEX_KEY = "com.index_feed";
+    public static        String TITLE_KEY(){return "com.visual.header";}
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        //**writeToIndexFile();**\\
+        setContentView(R.layout.activity_main_listing);
+        //Read Index Feed into JSON Array
+        try {
+            readMsgFeedFile(index_filename);
+            writeToIndexList();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        CustomListViewAdapter customAdapter = new CustomListViewAdapter(this, index_detail_setter);
+        // Assign adapter to ListView
+        setListAdapter(customAdapter);
+    }
+    //ONCE OFF CREATION OF EMPTY FILE COMMENTED OUT AFTER USE
+   /* public void writeToIndexFile(){
+        File msg_feed_file = new File(getExternalFilesDir(null), index_filename);
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(msg_feed_file);
+            OutputStreamWriter outputStreamWriter = new OutputStreamWriter(fileOutputStream);
+            outputStreamWriter.write("Test_Value");
+            outputStreamWriter.close();
+        }catch (IOException e){
+            Log.e("Exception", "File write failed: "+e.toString());
+        }
+    }*/
+
+    //Read file into JSON array
```

Comment:
Put a description of what method does inside the method. Can then be used to generate documentation.